tools/gen_wrf_io.c

```diff
@@ -85,7 +85,7 @@ gen_wrf_io2 ( FILE * fp , char * fname, char * structname , char * fourdname, no
   {
 
 
-    if ( p->ndims > 3 && ! p->node_kind & FOURD ) continue ; /* short circuit anything with more than 3 dims, (not counting 4d arrays) */
+    if ( p->ndims > 3 && ( (! p->node_kind) ) & FOURD ) continue ; /* short circuit anything with more than 3 dims, (not counting 4d arrays) */
```

Comment:
I believe the first set of inner parenthesis should group `(! p->node_kind) & FOURD`